### ISSUE TITLE
Static Library

### ISSUE THREAD

**Issue by hatter74** [wbenny/mini-tor#14]
Hi wbenny. Thank you for implementing a Tor client that is more Windows oriented. I'd like to use your code in a messaging app I'm developing, and would like to use mini-tor as a communication transport. Is there a way to compile your software into a static library for use in a C++ app? Thanks for the guidance

**Comment by wbenny** [comment]
Hi, in Visual Studio, you can change the project Configuration Type (`<ConfigurationType>` in .vcxproj) from "Application" to "Static Library".

**Comment by hatter74** [comment]
I seem to still be having some trouble getting the software to compile into a static library. I changed the Configuration Type to a Static Library, but Visual Studio is unhappy. I suspect that my issue isn't related to the static library compilation flag, but perhaps I'm missing something in my environment. I'm on Windows 10 using Visual Studio Community 2017 with Windows SDK 10.0.17763.0. I've posted the build output below. Any ideas?

(sorry for the wall of text)

```
1>------ Build started: Project: mini-tor, Configuration: Debug_Clang Win32 ------
1>base16.cpp
1>base64.cpp
1>main.cpp
1>console.cpp
1>In file included from mini\console.cpp:1:
1>In file included from mini/console.h:3:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/string.h:3:
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(11,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_offset_basis = 0x811C9DC5u;
1>  ^
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(12,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_prime        = 0x1000193u;
1>  ^
1>2 errors generated.
1>In file included from mini\crypto\capi\base64.cpp:1:
1>In file included from main.cpp:4:
1>In file included from mini\crypto\capi\base16.cpp:1:
1>In file included from mini\crypto\capi/base64.h:2:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/console.h:3:
1>In file included from mini\crypto\capi/base16.h:2:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/string.h:3:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/string.h:3:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/string.h:3:
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(11,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_offset_basis = 0x811C9DC5u;
1>  ^
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(11,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_offset_basis = 0x811C9DC5u;
1>  ^
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(12,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_prime        = 0x1000193u;
1>  ^
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(12,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_prime        = 0x1000193u;
1>  ^
1>2 errors generated.
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(11,3): error : 'inline' can only appear on functions
1>2 errors generated.
1>  inline constexpr size_type FNV_offset_basis = 0x811C9DC5u;
1>  ^
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(12,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_prime        = 0x1000193u;
1>  ^
1>2 errors generated.
1>base_encode.cpp
1>curve25519.cpp
1>random.cpp
1>random.cpp
1>In file included from mini\crypto\capi\detail\base_encode.cpp:1:
1>In file included from mini\crypto\capi\detail/base_encode.h:2:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/string.h:3:
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(11,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_offset_basis = 0x811C9DC5u;
1>  ^
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(12,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_prime        = 0x1000193u;
1>  ^
1>In file included from mini\crypto\capi\random.cpp:1:
1>In file included from mini\crypto\capi/random.h:2:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/byte_buffer.h:2:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/collections/list.h:3:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/byte_buffer_ref.h:3:
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(11,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_offset_basis = 0x811C9DC5u;
1>  ^
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(12,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_prime        = 0x1000193u;
1>  ^
1>In file included from mini\crypto\cng\random.cpp:1:
1>In file included from mini\crypto\cng/random.h:2:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/byte_buffer.h:2:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/collections/list.h:3:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/byte_buffer_ref.h:3:
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(11,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_offset_basis = 0x811C9DC5u;
1>  ^
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(12,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_prime        = 0x1000193u;
1>  ^
1>2 errors generated.
1>In file included from mini\crypto\cng\curve25519.cpp:1:
1>In file included from mini\crypto\cng/curve25519.h:4:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/byte_buffer.h:2:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/collections/list.h:3:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/byte_buffer_ref.h:3:
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(11,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_offset_basis = 0x811C9DC5u;
1>  ^
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(12,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_prime        = 0x1000193u;
1>  ^
1>base32.cpp
1>2 errors generated.
1>2 errors generated.
1>2 errors generated.
1>curve25519.cpp
1>file.cpp
1>file_enumerator.cpp
1>In file included from mini\crypto\ext\base32.cpp:1:
1>In file included from mini\crypto\ext/base32.h:2:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/string.h:3:
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(11,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_offset_basis = 0x811C9DC5u;
1>  ^
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(12,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_prime        = 0x1000193u;
1>  ^
1>In file included from mini\crypto\ext\curve25519.cpp:1:
1>In file included from mini\crypto\ext/curve25519.h:4:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/byte_buffer.h:2:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/collections/list.h:3:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/byte_buffer_ref.h:3:
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(11,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_offset_basis = 0x811C9DC5u;
1>  ^
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(12,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_prime        = 0x1000193u;
1>  ^
1>2 errors generated.
1>2 errors generated.
1>path.cpp
1>logger.cpp
1>In file included from mini\io\path.cpp:1:
1>In file included from mini\io/path.h:2:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/string.h:3:
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(11,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_offset_basis = 0x811C9DC5u;
1>  ^
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(12,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_prime        = 0x1000193u;
1>  ^
1>In file included from mini\io\file.cpp:1:
1>In file included from mini\io/file.h:4:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/string.h:3:
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(11,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_offset_basis = 0x811C9DC5u;
1>  ^
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(12,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_prime        = 0x1000193u;
1>  ^
1>In file included from mini\io\file_enumerator.cpp:1:
1>In file included from mini\io/file_enumerator.h:4:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/string.h:3:
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(11,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_offset_basis = 0x811C9DC5u;
1>  ^
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(12,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_prime        = 0x1000193u;
1>  ^
1>2 errors generated.
1>2 errors generated.
1>ssl_context.cpp
1>http.cpp
1>2 errors generated.
1>ssl_socket.cpp
1>In file included from mini\net\detail\ssl_context.cpp:1:
1>In file included from mini\net\detail/ssl_context.h:2:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/net/tcp_socket.h:2:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/string.h:3:
1>In file included from mini\net\http.cpp:1:
1>In file included from mini\net/http.h:2:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/string.h:3:
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(11,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_offset_basis = 0x811C9DC5u;
1>  ^
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(12,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_prime        = 0x1000193u;
1>  ^
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(11,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_offset_basis = 0x811C9DC5u;
1>  ^
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(12,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_prime        = 0x1000193u;
1>  ^
1>In file included from mini\logger.cpp:2:
1>In file included from mini/console.h:3:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/string.h:3:
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(11,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_offset_basis = 0x811C9DC5u;
1>  ^
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(12,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_prime        = 0x1000193u;
1>  ^
1>2 errors generated.
1>ssl_stream.cpp
1>In file included from mini\net\ssl_socket.cpp:1:
1>In file included from mini\net/ssl_socket.h:2:
1>In file included from mini\net/tcp_socket.h:2:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/string.h:3:
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(11,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_offset_basis = 0x811C9DC5u;
1>  ^
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(12,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_prime        = 0x1000193u;
1>  ^
1>In file included from mini\net\ssl_stream.cpp:1:
1>In file included from mini\net/ssl_stream.h:2:
1>In file included from mini\net/tcp_socket.h:2:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/string.h:3:
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(11,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_offset_basis = 0x811C9DC5u;
1>  ^
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(12,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_prime        = 0x1000193u;
1>  ^
1>2 errors generated.
1>tcp_socket.cpp
1>2 errors generated.
1>string.cpp
1>In file included from mini\net\tcp_socket.cpp:1:
1>In file included from mini\net/tcp_socket.h:2:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/string.h:3:
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(11,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_offset_basis = 0x811C9DC5u;
1>  ^
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(12,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_prime        = 0x1000193u;
1>  ^
1>2 errors generated.
1>string_ref.cpp
1>In file included from mini\string.cpp:1:
1>In file included from mini/string.h:3:
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(11,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_offset_basis = 0x811C9DC5u;
1>  ^
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(12,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_prime        = 0x1000193u;
1>  ^
1>2 errors generated.
1>event.cpp
1>2 errors generated.
1>thread.cpp
1>In file included from mini\string_ref.cpp:1:
1>In file included from mini/string_ref.h:2:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/string.h:3:
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(11,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_offset_basis = 0x811C9DC5u;
1>  ^
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(12,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_prime        = 0x1000193u;
1>  ^
1>2 errors generated.
1>2 errors generated.
1>time.cpp
1>cell.cpp
1>In file included from mini\time.cpp:1:
1>In file included from mini/time.h:2:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/string.h:3:
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(11,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_offset_basis = 0x811C9DC5u;
1>  ^
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(12,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_prime        = 0x1000193u;
1>  ^
1>In file included from mini\tor\cell.cpp:1:
1>In file included from mini\tor/cell.h:2:
1>In file included from mini\tor/common.h:2:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/string.h:3:
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(11,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_offset_basis = 0x811C9DC5u;
1>  ^
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(12,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_prime        = 0x1000193u;
1>  ^
1>In file included from mini\threading\event.cpp:1:
1>In file included from mini\threading/event.h:5:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/time.h:2:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/string.h:3:
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(11,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_offset_basis = 0x811C9DC5u;
1>  ^
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(12,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_prime        = 0x1000193u;
1>  ^
1>In file included from mini\threading\thread.cpp:1:
1>In file included from mini\threading/thread.h:5:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/time.h:2:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/string.h:3:
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(11,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_offset_basis = 0x811C9DC5u;
1>  ^
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(12,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_prime        = 0x1000193u;
1>  ^
1>2 errors generated.
1>2 errors generated.
1>circuit.cpp
1>circuit_node_crypto_state.cpp
1>2 errors generated.
1>In file included from mini\tor\circuit.cpp:1:
1>In file included from mini\tor/circuit.h:2:
1>In file included from mini\tor/tor_socket.h:2:
1>In file included from mini\tor/onion_router.h:3:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/byte_buffer.h:2:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/collections/list.h:3:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/byte_buffer_ref.h:3:
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(11,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_offset_basis = 0x811C9DC5u;
1>  ^
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(12,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_prime        = 0x1000193u;
1>  ^
1>consensus.cpp
1>In file included from mini\tor\circuit_node_crypto_state.cpp:1:
1>In file included from mini\tor/circuit_node_crypto_state.h:2:
1>In file included from mini\tor/cell.h:2:
1>In file included from mini\tor/common.h:2:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/string.h:3:
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(11,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_offset_basis = 0x811C9DC5u;
1>  ^
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(12,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_prime        = 0x1000193u;
1>  ^
1>2 errors generated.
1>hybrid_encryption.cpp
1>In file included from mini\tor\consensus.cpp:1:
1>In file included from mini\tor/consensus.h:2:
1>In file included from mini\tor/onion_router.h:3:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/byte_buffer.h:2:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/collections/list.h:3:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/byte_buffer_ref.h:3:
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(11,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_offset_basis = 0x811C9DC5u;
1>  ^
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(12,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_prime        = 0x1000193u;
1>  ^
1>In file included from mini\tor\crypto\hybrid_encryption.cpp:1:
1>In file included from mini\tor\crypto/hybrid_encryption.h:2:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/byte_buffer.h:2:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/collections/list.h:3:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/byte_buffer_ref.h:3:
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(11,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_offset_basis = 0x811C9DC5u;
1>  ^
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(12,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_prime        = 0x1000193u;
1>  ^
1>2 errors generated.
1>2 errors generated.
1>key_agreement_tap.cpp
1>circuit_node.cpp
1>2 errors generated.
1>key_agreement_ntor.cpp
1>In file included from mini\tor\circuit_node.cpp:1:
1>In file included from mini\tor/circuit_node.h:2:
1>In file included from mini\tor/circuit_node_crypto_state.h:2:
1>In file included from mini\tor/cell.h:2:
1>In file included from mini\tor/common.h:2:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/string.h:3:
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(11,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_offset_basis = 0x811C9DC5u;
1>  ^
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(12,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_prime        = 0x1000193u;
1>  ^
1>In file included from mini\tor\crypto\key_agreement_tap.cpp:1:
1>In file included from mini\tor\crypto/key_agreement_tap.h:2:
1>In file included from mini\tor\crypto/key_agreement.h:2:
1>In file included from mini\tor\crypto/../onion_router.h:3:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/byte_buffer.h:2:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/collections/list.h:3:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/byte_buffer_ref.h:3:
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(11,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_offset_basis = 0x811C9DC5u;
1>  ^
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(12,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_prime        = 0x1000193u;
1>  ^
1>2 errors generated.
1>hidden_service.cpp
1>In file included from mini\tor\crypto\key_agreement_ntor.cpp:1:
1>In file included from mini\tor\crypto/key_agreement_ntor.h:2:
1>In file included from mini\tor\crypto/key_agreement.h:2:
1>In file included from mini\tor\crypto/../onion_router.h:3:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/byte_buffer.h:2:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/collections/list.h:3:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/byte_buffer_ref.h:3:
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(11,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_offset_basis = 0x811C9DC5u;
1>  ^
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(12,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_prime        = 0x1000193u;
1>  ^
1>In file included from mini\tor\hidden_service.cpp:1:
1>In file included from mini\tor/hidden_service.h:2:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/stack_buffer.h:3:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/byte_buffer_ref.h:3:
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(11,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_offset_basis = 0x811C9DC5u;
1>  ^
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(12,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_prime        = 0x1000193u;
1>  ^
1>2 errors generated.
1>2 errors generated.
1>onion_router.cpp
1>consensus_parser.cpp
1>2 errors generated.
1>hidden_service_descriptor_parser.cpp
1>In file included from mini\tor\parsers\consensus_parser.cpp:1:
1>In file included from mini\tor\parsers/consensus_parser.h:2:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/time.h:2:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/string.h:3:
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(11,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_offset_basis = 0x811C9DC5u;
1>  ^
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(12,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_prime        = 0x1000193u;
1>  ^
1>In file included from mini\tor\onion_router.cpp:1:
1>In file included from mini\tor/onion_router.h:3:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/byte_buffer.h:2:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/collections/list.h:3:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/byte_buffer_ref.h:3:
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(11,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_offset_basis = 0x811C9DC5u;
1>  ^
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(12,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_prime        = 0x1000193u;
1>  ^
1>2 errors generated.
1>introduction_point_parser.cpp
1>In file included from mini\tor\parsers\hidden_service_descriptor_parser.cpp:1:
1>In file included from mini\tor\parsers/hidden_service_descriptor_parser.h:2:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/string.h:3:
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(11,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_offset_basis = 0x811C9DC5u;
1>  ^
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(12,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_prime        = 0x1000193u;
1>  ^
1>In file included from mini\tor\parsers\introduction_point_parser.cpp:1:
1>In file included from mini\tor\parsers/introduction_point_parser.h:2:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/string.h:3:
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(11,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_offset_basis = 0x811C9DC5u;
1>  ^
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(12,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_prime        = 0x1000193u;
1>  ^
1>2 errors generated.
1>2 errors generated.
1>onion_router_descriptor_parser.cpp
1>relay_cell.cpp
1>2 errors generated.
1>tor_socket.cpp
1>In file included from mini\tor\parsers\onion_router_descriptor_parser.cpp:1:
1>In file included from mini\tor\parsers/onion_router_descriptor_parser.h:2:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/string.h:3:
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(11,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_offset_basis = 0x811C9DC5u;
1>  ^
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(12,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_prime        = 0x1000193u;
1>  ^
1>In file included from mini\tor\relay_cell.cpp:1:
1>In file included from mini\tor/relay_cell.h:2:
1>In file included from mini\tor/cell.h:2:
1>In file included from mini\tor/common.h:2:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/string.h:3:
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(11,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_offset_basis = 0x811C9DC5u;
1>  ^
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(12,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_prime        = 0x1000193u;
1>  ^
1>2 errors generated.
1>tor_stream.cpp
1>In file included from mini\tor\tor_socket.cpp:1:
1>In file included from mini\tor/tor_socket.h:2:
1>In file included from mini\tor/onion_router.h:3:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/byte_buffer.h:2:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/collections/list.h:3:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/byte_buffer_ref.h:3:
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(11,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_offset_basis = 0x811C9DC5u;
1>  ^
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(12,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_prime        = 0x1000193u;
1>  ^
1>In file included from mini\tor\tor_stream.cpp:1:
1>In file included from mini\tor/tor_stream.h:2:
1>In file included from mini\tor/common.h:2:
1>In file included from V:\Code\Libraries\mini-tor-0.1.22\mini/string.h:3:
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(11,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_offset_basis = 0x811C9DC5u;
1>  ^
1>V:\Code\Libraries\mini-tor-0.1.22\mini/hash.h(12,3): error : 'inline' can only appear on functions
1>  inline constexpr size_type FNV_prime        = 0x1000193u;
1>  ^
1>2 errors generated.
1>2 errors generated.
1>2 errors generated.
1>2 errors generated.
1>Done building project "mini-tor.vcxproj" -- FAILED.
```

**Comment by wbenny** [comment]
Use MSVC, not clang (use any other project configuration, that doesn't end with `_Clang`).
